### ISSUE TITLE
Skinsrestorer & floodgate v2 onjoin

### ISSUE THREAD

**Issue by xknat** [SkinsRestorer/SkinsRestorerX#898]
align floodgate with skinsrestorer to not run onjoin when floodgate is already handeling the skin and override it when the bedrock player does use the skinsrestorer skin set.

**Comment by ExtremeFiretop** [comment]
Hey @xknat 

Just curious if this is the reason that SkinRestorer works with Floodgate on every second login for me?
As in; if I just booted the server 5 or 10 minutes ago and a Geyser/Floodgate user logs in, they will have no skin.

However if they instantly log out and login again their skin appears as normal.
I figured this may be what you are referencing but I am not sure of the symptoms of this bug you are describing above.

**Comment by xknat** [comment]
> Hey @xknat
> 
> Just curious if this is the reason that SkinRestorer works with Floodgate on every second login for me? As in; if I just booted the server 5 or 10 minutes ago and a Geyser/Floodgate user logs in, they will have no skin.
> 
> However if they instantly log out and login again their skin appears as normal. I figured this may be what you are referencing but I am not sure of the symptoms of this bug you are describing above.

could be yeah, we did not check on their new version changes and event's may no longer match to make skins always work.

**Comment by ExtremeFiretop** [comment]
> > Hey @xknat
> > Just curious if this is the reason that SkinRestorer works with Floodgate on every second login for me? As in; if I just booted the server 5 or 10 minutes ago and a Geyser/Floodgate user logs in, they will have no skin.
> > However if they instantly log out and login again their skin appears as normal. I figured this may be what you are referencing but I am not sure of the symptoms of this bug you are describing above.
> 
> could be yeah, we did not check on their new version changes and event's may no longer match to make skins always work.

It's very low priority; I am sure you have many other things that will take up more of your precious time.
It was simply one of the things I noticed after updating to the dev build of Floodgate and SkinRestorer.

Most of my users don't even notice or care; if they do I just tell them to try logging in again and it works every time lol.
Thanks for your time and explanation. I appreciate it.

**Comment by xknat** [comment]
waiting on https://github.com/GeyserMC/Floodgate/issues/426

**Comment by xknat** [comment]
Floodgate fixed the event on their end in the build - 68

**Comment by AlexProgrammerDE** [comment]
Should be resolved by v15. May need some extra future testing to find possible edge cases.
